disasm-test/known_bugs/p2.cpp

```diff
@@ -0,0 +1,10 @@
+##> rootMatchName: p2.cpp
+##> summary: failure to llvm-as the llvm-disasm results
+
+Using this module's llvm-disasm to convert the clang++ generated bitcode into assembly (.ll), the resulting assembly is invalid for llvm-as:
+
+    error: expected type
+    define linkonce_ddr default i32 @_ZN4uorb14uorb_advertiseEPK21orb_metadata(%class.uorb* %32
```

Comment:
This error is just a typo, it should be linkonce_odr. The data member is named correctly, the error is here: https://github.com/elliottt/llvm-pretty/blob/8124fc0265b6ca5fde5812c789f40b2ea54af678/src/Text/LLVM/PP.hs#L463

With this fixed, there is an instance of #222 which is probably the same cause.

Reply:
I reduced p2.cpp (for #222) to
```
struct h {
  bool a;
};

void g(struct h *s) {
}

void f() {
  struct h s;
}
```

I do not have a fix yet, but the bitcode is now small enough to walk through manually.

Reply:
Great sleuthing!

I actually think the culprit here isn't #222, but rather #260, as the diff failure is caused by improper inlining. I'll add this example to the comments in #260.